### PR TITLE
Avoid %% escape in xtrigger func args (master).

### LINKED ISSUE #3256: Allow proper escaping in xtrigger format statements
I had a format string that I wanted to pass through the xtrigger formatting, but the regular expression used to match format strings and check their validity did not honor the Python escape sequence, i.e.
"%%(asdf)s" after running through string format is "%(asdf)s", but the current xtrigger format check in the suite validation complains that "asdf" isn't one of the approved variables for string substitution.

Use case here is to allow for parameterized filenames in the xtrigger, but I can't directly use cycle points as the files have 10-digit YYYYMMDDHH in the name so I resorted to passing in a format-string filename and a date-time-group format.

### PR DESCRIPTION
**Pull request by hjoliver**
This is the master branch (cylc-8) companion of #3257 - see documentation there.

These changes close #3256 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).

- [x] Appropriate tests are included (unit and/or functional).

- [x] Includes an appropriate entry in the release change log `CHANGES.md`.

- [x] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/50

### CODE REVIEW

File: cylc/flow/tests/test_xtrigger_mgr.py [hunk 1]
```diff
@@ -14,551 +16,24 @@
...
-from cylc.flow.broadcast_mgr import BroadcastMgr
-from cylc.flow.cycling.iso8601 import ISO8601Point, init
-from cylc.flow.subprocctx import SubFuncContext
-from cylc.flow.subprocpool import SubProcPool
-from cylc.flow.task_proxy import TaskProxy
-from cylc.flow.taskdef import TaskDef
-from cylc.flow.xtrigger_mgr import XtriggerManager
-
-
-def test_constructor():
```
Comment (kinow) [inline cylc/flow/tests/test_xtrigger_mgr.py]: @hjoliver I think while merging it you accidentally removed all the tests from this file? I think we should only include your new one instead

Comment (hjoliver) [reply cylc/flow/tests/test_xtrigger_mgr.py]: Oops yes. There were no unit tests for this module on 7.8.x so I created the associated test file from scratch, and didn't check to see if it already existed on master before copying over.  (So I replaced a whole bunch of existing unit tests with one new one - doh). Will fix later today...

Comment (hjoliver) [reply cylc/flow/tests/test_xtrigger_mgr.py]: Fixed!

**Comment by kinow** [comment]
Kicked Travis functional test, once it passes will merge it :tada:

**Pull request merged**